### PR TITLE
add more SmartOrderStrategy test cases

### PR DESCRIPTION
**Pull request by charlesjhongc**
Added some cases
- WETH unwrap handling
- multiple AMMs
- with V6 RFQ
- with V6 LimitOrderSwap

### CODE REVIEW

File: test/forkMainnet/SmartOrderStrategy/IntegrationV6.t.sol [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    address maker = vm.addr(makerPrivateKey);
+    bytes defaultPermit = abi.encodePacked(TokenCollector.Source.Token);
+    uint256 defaultSalt = 1234;
+    RFQ rfq;
+    LimitOrderSwap limitOrderSwap;
+
+    function setUp() public override {
+        super.setUp();
+
+        rfq = new RFQ(owner, UNISWAP_PERMIT2_ADDRESS, address(0), IWETH(WETH_ADDRESS), payable(owner));
```
[the remark this answers is not in the document]
Comment (charlesjhongc) [reply test/forkMainnet/SmartOrderStrategy/IntegrationV6.t.sol]: fixed in f671510

File: test/forkMainnet/SmartOrderStrategy/IntegrationV6.t.sol [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        });
+        RFQTx memory rfqTx = RFQTx({ rfqOffer: rfqOffer, takerRequestAmount: rfqOffer.takerTokenAmount, recipient: payable(address(smartOrderStrategy)) });
+        bytes memory makerSig = _signRFQOffer(makerPrivateKey, rfqOffer);
+        bytes memory rfqData = abi.encodeWithSelector(RFQ_FILL_SELECTOR, rfqTx, makerSig, defaultPermit, defaultPermit);
+
+        ISmartOrderStrategy.Operation[] memory operations = new ISmartOrderStrategy.Operation[](1);
+        operations[0] = ISmartOrderStrategy.Operation({
+            dest: address(rfq),
+            inputToken: rfqOffer.takerToken,
+            inputRatio: 0,
```
[the remark this answers is not in the document]
Comment (charlesjhongc) [reply test/forkMainnet/SmartOrderStrategy/IntegrationV6.t.sol]: added in f671510

File: test/forkMainnet/SmartOrderStrategy/IntegrationV6.t.sol [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            extraAction: bytes(""),
+            takerTokenPermit: defaultPermit
+        });
+        bytes memory loData = abi.encodeWithSelector(LimitOrderSwap.fillLimitOrderFullOrKill.selector, order, makerSig, takerParams);
+
+        ISmartOrderStrategy.Operation[] memory operations = new ISmartOrderStrategy.Operation[](1);
+        operations[0] = ISmartOrderStrategy.Operation({
+            dest: address(limitOrderSwap),
+            inputToken: order.takerToken,
+            inputRatio: 0,
```
[the remark this answers is not in the document]
Comment (charlesjhongc) [reply test/forkMainnet/SmartOrderStrategy/IntegrationV6.t.sol]: added in f671510

File: test/forkMainnet/SmartOrderStrategy/AMMs.t.sol [hunk 4]
```diff
@@ -112,7 +27,7 @@ contract SmartOrderStrategyTest is Test, Tokens, BalanceUtil {
         ISmartOrderStrategy.Operation[] memory operations = new ISmartOrderStrategy.Operation[](1);
         operations[0] = ISmartOrderStrategy.Operation({
             dest: UNISWAP_V2_ADDRESS,
-            inputToken: defaultInputToken,
+            inputToken: address(0),
```
[the remark this answers is not in the document]
Comment (charlesjhongc) [reply test/forkMainnet/SmartOrderStrategy/AMMs.t.sol]: Actually this address doesn't matter if no amount replacement will be made. But I changed it as input token address for the sake of readability.

**Review by NIC619 — APPROVED** [review]

**Review by pilagod — APPROVED** [review]

**Pull request merged**